src/rust/aggregator/src/data/leaderboards.rs

```diff
@@ -0,0 +1,261 @@
+                                WHERE place_market_order_events.txn_version = competition_indexed_events.txn_version
+                                AND place_market_order_events.event_idx = competition_indexed_events.event_idx
+                                AND competition_indexed_events.competition_id = $2
+                            )
+                            AND time > $3 AND time < $4
+                            UNION
+                            SELECT DISTINCT signing_account as "user", integrator FROM place_swap_order_events
+                            WHERE NOT EXISTS (
+                                SELECT * FROM competition_indexed_events
+                                WHERE place_swap_order_events.txn_version = competition_indexed_events.txn_version
+                                AND place_swap_order_events.event_idx = competition_indexed_events.event_idx
+                                AND competition_indexed_events.competition_id = $2
+                            )
+                            AND time > $3 AND time < $4
+                        ) AS a
+                        WHERE integrator IN (SELECT unnest(frontends_required) FROM competition_metadata)
+                        AND "user" = $1
+                    ),
+                    trades = (
+                        WITH homogenous_fills AS (
+                            SELECT taker_address AS "user", taker_order_id AS order_id
+                            FROM fill_events
+                            WHERE time > $3 AND time < $4
+                            UNION
+                            SELECT maker_address AS "user", maker_order_id AS order_id
+                            FROM fill_events
+                            WHERE time > $3 AND time < $4
+                        )
+                        SELECT COUNT(order_id) AS orders
+                        FROM homogenous_fills
+                        WHERE "user" = $1
+                    )
+                WHERE "user" = $1
+            "#,
+            user.user,
+            comp.id,
+            comp.start,
+            comp.end,
+        )
+        .execute(transaction as &mut PgConnection)
+        .await
+        .map_err(|e| DataAggregationError::ProcessingError(anyhow!(e)))?;
+    }
+    sqlx::query!(
+        r#"
+            INSERT INTO competition_indexed_events SELECT txn_version, event_idx, $1 FROM fill_events
+            WHERE NOT EXISTS (
+                SELECT * FROM competition_indexed_events
+                WHERE fill_events.txn_version = competition_indexed_events.txn_version
+                AND fill_events.event_idx = competition_indexed_events.event_idx
+                AND competition_indexed_events.competition_id = $1
+            )
+            AND fill_events.time > $2 AND fill_events.time < $3
+        "#,
+        comp.id,
+        comp.start,
+        comp.end,
+    )
+    .execute(transaction as &mut PgConnection)
```

Comment:
How about abstracting this out a bit too?

In the user history aggregation it's clear what's going on

But here there is so much nested SQL that I find it hard to immediately grasp

Comments for each query will help too